lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -28,18 +28,13 @@ def consume_batch(payloads, metadata)
           zip(metadata[:keys]).
           map { |p, k| Deimos::Message.new(p, nil, key: k) }
 
-        tags = %W(topic:#{metadata[:topic]})
-
-        Deimos.instrument('ar_consumer.consume_batch', tags) do
-          # The entire batch should be treated as one transaction so that if
-          # any message fails, the whole thing is rolled back or retried
-          # if there is deadlock
-          Deimos::Utils::DeadlockRetry.wrap(tags) do
-            if @compacted || self.class.config[:no_keys]
-              update_database(compact_messages(messages))
-            else
-              uncompacted_update(messages)
-            end
+        @tags = %W(topic:#{metadata[:topic]})
```

Comment:
I think it's weird to pass metric tags so far down into the workflow. Can we do something like

```
Deimos.config.tracer.active_span.set_tag("topic", metadata[:topic])
...
Deimos.config.tracer.active_span.get_tag("topic")
```

We'd have to add a `get_tag` method to the `provider` class but it feels a lot better because we can separate out the tracing/metric logic from the actual business logic.